### PR TITLE
Fix s3 listBuckets rest api non-standard response issue

### PR DESCRIPTION
**Pull request by jffree**
### What changes are proposed in this pull request?

Fix s3 listBuckets rest api non-standard response issue.

### Why are the changes needed?

According to [AWS S3 ListObject API](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjects.html), `CommonPrefixes` in `listBuckets` response  should be an array. 

Like this:

```
<ListBucketResult>
  <Contents>
    <LastModified>2021-09-22T15:17:39.579Z</LastModified>
    <Key>key1</Key>
    <Size>27040</Size>
  </Contents>
  <Contents>
    <LastModified>2021-09-22T15:17:41.463Z</LastModified>
    <Key>key2</Key>
    <Size>27040</Size>
  </Contents>
  <KeyCount>13</KeyCount>
  <MaxKeys>1000</MaxKeys>
  <Delimiter>/</Delimiter>
  <EncodingType>url</EncodingType>
  <CommonPrefixes>
    <Prefix>fake_1/</Prefix>
  </CommonPrefixes>
  <CommonPrefixes>
    <Prefix>fake_2/</Prefix>
  </CommonPrefixes>
  <IsTruncated>false</IsTruncated>
  <Name>testbucket</Name>
</ListBucketResult>
```

Update s3 listBucket rest api doc.

### Does this PR introduce any user facing changes?

no

### CODE REVIEW

File: core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java [hunk 1]
```diff
@@ -63,7 +63,7 @@
   private List<Content> mContents;
 
   // List of common prefixes (aka. folders)
-  private CommonPrefixes mCommonPrefixes;
+  private List<CommonPrefixes> mCommonPrefixes;
```
Comment (HelloHorizon) [inline core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java:66]: Also change the comment here?

**Comment by ZacBlanco** [comment]
@jffree Are you attempting to support the original ListObjects action (linked in your PR description)?

The API is meant to support the [ListObjectsV2 endpoint](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html). The original ListObjects request/response is deprecated and only supported for legacy applications by AWS.

File: core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java [hunk 2]
```diff
@@ -218,27 +212,27 @@ public String getNextMarker() {
    * @return the common prefixes
    */
   @JacksonXmlProperty(localName = "CommonPrefixes")
-  public CommonPrefixes getCommonPrefixes() {
+  @JacksonXmlElementWrapper(useWrapping = false)
```
Comment (ZacBlanco) [inline core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java:215]: why add `useWrapping = false` here?

File: docs/en/api/S3-API.md [hunk 3]
```diff
@@ -139,16 +139,16 @@ Server: Jetty(9.2.z-SNAPSHOT)
...
+Content-Length: 306
+Server: Jetty(9.4.43.v20210629)
 
-<ListBucketResult><Name>/testbucket</Name><Prefix/><ContinuationToken/><NextContinuationToken/><KeyCount>1</KeyCount><MaxKeys>1000</MaxKeys><IsTruncated>false</IsTruncated><Contents><Key>testobject</Key><LastModified>2019-06-18T14:24:33.029Z</LastModified><ETag></ETag><Size>27040</Size><StorageClass>STANDARD</StorageClass></Contents></ListBucketResult>
+<ListBucketResult><Contents><LastModified>2021-09-22T15:14:40.754Z</LastModified><Key>testobject</Key><Size>27040</Size></Contents><KeyCount>1</KeyCount><MaxKeys>1000</MaxKeys><Delimiter>/</Delimiter><EncodingType>url</EncodingType><IsTruncated>false</IsTruncated><Name>testbucket</Name></ListBucketResult>
 ```
 
 #### Listing a bucket with multiple objects
-You can upload more files and use the `max-keys` and `continuation-token` as the [GET bucket request parameter](https://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html). For example:
+You can upload more files and use the `max-keys` and `marker` as the [GET bucket request parameter](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjects.html). For example:
```
Comment (ZacBlanco) [inline docs/en/api/S3-API.md:151]: please reference ListObjectsV2 here

**Review by ZacBlanco — CHANGES_REQUESTED** [review]
Overall this is good. Just two minor comments. Thanks for fixing the API conformance.

**Review by HelloHorizon — CHANGES_REQUESTED** [review]
@jffree Thanks for the improvement! Left one minor comment.

File: core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java (hunk 1 above)
Comment (jffree) [reply core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java:66]: Comment `List of common prefixes` is right.  I don't think it needs to be modified.

Like comment of `mContents`:
```
  // List of files.
  private List<Content> mContents;
```

File: docs/en/api/S3-API.md (hunk 3 above)
Comment (jffree) [reply docs/en/api/S3-API.md:151]: `marker` is implemented in `listBuckets` interface, but `continuation-token` is not implemented.
`ListObjectsV2` did not contains `marker`, only `API_ListObjects` contains.

File: core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java (hunk 2 above)
Comment (jffree) [reply core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java:215]: Refer to the annotations of `mContents`.

```
  /**
   * @return the list of contents
   */
  @JacksonXmlProperty(localName = "Contents")
  @JacksonXmlElementWrapper(useWrapping = false)
  public List<Content> getContents() {
    return mContents;
  }
```

Through testing, found that `useWrapping = false` is not necessary and can be deleted.

File: docs/en/api/S3-API.md (hunk 3 above)
Comment (jffree) [reply docs/en/api/S3-API.md:151]: @ZacBlanco @HelloHorizon 
Maybe we can open a new `type-feature` to support `ListObjectsV2` ?

Comment (ZacBlanco) [reply docs/en/api/S3-API.md:151]: I see. Yes, we should open an issue for this. Thanks for pointing it out.

**Review by ZacBlanco — APPROVED** [review]

**Review by HelloHorizon — APPROVED** [review]

**Comment by HelloHorizon** [comment]
alluxio-bot, merge this please

**Comment by HelloHorizon** [comment]
alluxio-bot, cherry-pick this to branch-2.6 please

**Pull request merged**